sunkit_instruments/suvi/io.py

```diff
@@ -0,0 +1,320 @@
+import os
+
+import h5py
+import numpy
+
+from astropy import units as u
+from astropy.io import fits
+from astropy.time import Time
+
+from sunkit_instruments.suvi import fix_L1b_header
+
+__all__ = ["read_suvi"]
+
+def read_suvi(input_filename, return_DQF=False, return_header_only=False):
+    """
+    Read a SUVI L1b FITS or netCDF file or a L2 HDR composite FITS file.
+    Return data and header, optionally the data quality flag array (DQF)
+    for L1b files. For SUVI L1b FITS files, the broken FITS header is
+    fixed automatically (broken because of the wrong implementation of
+    the CONTINUE convention). This read function is intented to provide
+    a consistent file interface for FITS and netCDF, L1b and L2.
+
+    .. note::
+        The type of file is determined by pattern matching in the
+        filenames, e.g. "-L1b-Fe171" for a 171 L1b file and "-l2-ci171"
+        for a 171 L2 HDR composite. If those patterns are not found
+        in the filename, the files will not be recognized.
+
+    .. note::
+        If input_filename is an L1b netCDF file, the information from
+        the netCDF file is transformed into a FITS header.
+
+    Parameters
+    ----------
+    input_filename: `str`
+        File to read.
+
+    return_DQF: `bool`, optional. Default: False.
+        If True, also returns the data quality flag array (DQF, only for L1b files).
+
+    return_header_only: `bool`, optional. Default: False.
+        If True, does not return the data array (but optionally the DQF).
+
+    Returns
+    -------
+    header: `astropy.io.fits.header.Header`
+        header, if return_header_only is True.
+
+    header, data: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data.
+
+    header, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data quality flags, if both return_header_only and return_DQF
+        are True.
+
+    header, data, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`, `~numpy.ndarray`
+        header, data, and data quality flags if return_DQF is True.
+    """
+    fits_file_extensions = ('.fits', '.fts', '.fits.gz', 'fts.gz', 'fits.bz', 'fts.bz')
+    netCDF_file_extensions = ('.nc', '.nc.gz', '.nc.bz', '.cdf', '.cdf.gz', '.cdf.bz')
+
+    composite_matches = ['-l2-ci094', '-l2-ci131', '-l2-ci171', '-l2-ci195', '-l2-ci284', '-l2-ci304']
+    L1b_matches = ['-L1b-Fe093', '-L1b-Fe131', '-L1b-Fe171', '-L1b-Fe195', '-L1b-Fe284', '-L1b-He303']
```

Comment:
Since these are fixed constants, I would like them moved to the top of the file, under the imports and then be renamed to be all uppercase.

Reply:
Ok.